File: ruby/resistor-color-duo/resistor_color_duo.rb
=begin
ResistorColorDuo allows you to determine the value of a resistor from the colored bands.
=end

class ResistorColorDuo

  COLORS = ["black", "brown", "red","orange","yellow", "green", "blue","violet","grey","white"]

  def self.value(colors)
    #the first color is the tens digit, the second is the ones digit.  All additional colors will be ignored
    COLORS.index(colors[0]) * 10 + COLORS.index(colors[1])
  end
end

File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
class ResistorColorDuo {

    private String[] resistor_code = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};

    int value(String[] colors) {
        return find_index(colors[0])*10 + find_index(colors[1]);
    }

    int find_index(String search){
        for (int i = 0; i<resistor_code.length; i++){
            if (search == resistor_code[i]){
                return i;
            }
        }
        return -1;
    }
}
